src/analysis/utils/disk-utils.ts

```diff
@@ -62,7 +62,8 @@ export const getCurrentAttachedDataDisk = (
   app: App,
   appDataDisks: PersistentDisk[]
 ): DecoratedPersistentDisk | undefined => {
-  const currentDisk: PersistentDisk | undefined = _.find({ name: app?.diskName }, appDataDisks);
+  const currentDisk: PersistentDisk | undefined =
+    !!app && app.diskName === null ? undefined : appDataDisks.find(({ name }) => app.diskName === name);
```

Comment:
Is the `!!app` needed here? This function as typed as requiring that `app` be an `App`, not `App | undefined`.

If `app` can actually be undefined, it looks like that is not handled, since the `find` would attempt to read `app.diskName`.

Reply:
Arg, I think the original typing in the function is wrong, and the App can be undefined :( Let me take another pass at handling that case